### DISCUSSION TITLE
setting the colour of a button

### DISCUSSION THREAD

**Discussion by norricorp** [slint-ui/slint#4902] [Q&A]
Am I right to think that a Button inherits its colour from the background of the element eg Rectangle it is sitting in?
Or can I set the colour of a Button? And the text colour?

I can see in Custom Widgets, that I can create my own button by inheriting rectangle (which does have a background colour) but then of course I have to set all of the other properties of a button too.

So is that code the way forwards?

**Comment by ogoffart** [comment]
The button from the std-widgets.slint are meant to look like "native" buttons. And you can't change their appearance.
If you want a custom appearance, you can do a custom Button.

**Comment by syf20020816** [comment]
Perhaps SurrealismUI's `SButton` can meet your needs
https://github.com/Surrealism-All/SurrealismUI
SButton: [address for SButton](https://github.com/Surrealism-All/SurrealismUI/wiki/06_Basic-Components-%E5%9F%BA%E7%A1%80%E7%BB%84%E4%BB%B6#sbutton-%E6%8C%89%E9%92%AE)

**Reply by norricorp** [reply]
Thank you for this - very interesting
